test/test_framework.py

```diff
@@ -889,9 +897,13 @@ class SomeObject(Object):
             # it still contains the old value at this point. State is overridden here via save_snapshot to validate that
             # the modification will not be saved when StoredStateData is not dirty. This check assumes that the artificially
             # created StoredStateData does not observe the on_commit event.
+            # This is the one place where we intentionally create a second object for the framework.
```

Comment:
Tests seem to have been adapted for the new changes, but apparently we're not covering the new requirement in a way that breaking it would raise an alarm. Can we please have some specific testing added to ensure the logic is sane, assuming it's not just me missing it.

Reply:
You're absolutely correct, thanks for the catch.